projects/components/table/src/table.component.ts

```diff
@@ -0,0 +1,390 @@
+    for (const key of currentParams.keys) {
+      if (key !== this.tableId) {
+        newQueryParams[key] = currentParams.get(key);
+      }
+    }
+
+    this.router.navigate([], {
+      queryParams: newQueryParams,
+      relativeTo: this.route,
+    });
+
+    this.flipContainer.toggleFlip();
+  }
+
+  private requestUpdate() {
+    const newQueryParams: { [id: string]: any } = {};
+
+    const currentParams = this.route.snapshot.queryParamMap;
+    for (const key of currentParams.keys) {
+      newQueryParams[key] = currentParams.get(key);
+    }
+
+    if (this.tableId) {
+      const updateInfo = this.dataSource.getUpdateDataInfo();
+      newQueryParams[this.tableId] = asQueryParams(updateInfo);
+    }
+
+    this.router.navigate([], {
+      queryParams: newQueryParams,
+      relativeTo: this.route,
+    });
+  }
+
+  private initListSettingsSubscription() {
+    const tableSettings$ = this.settingsService.settings$.pipe(
+      map(settings => {
+        if (this.tableId && settings && settings[this.tableId]) {
+          return settings[this.tableId];
+        }
+        return null;
+      })
+    );
+    const urlSettings$ = this.route.queryParamMap.pipe(
+      map(queryParamMap => {
+        if (this.tableId && queryParamMap.has(this.tableId)) {
+          return fromQueryParams(queryParamMap.get(this.tableId));
+        }
+        return null;
+      })
+    );
+    this.subscriptions.push(
+      combineLatest([urlSettings$, tableSettings$, this.settingsService.defaultPageSize$])
+        .pipe(
+          // guards agains multiple data updates, when multiple emits happen at the same time.
+          debounceTime(0)
+        )
+        .subscribe({
+          next: ([urlSettings, savedSettings, defaultPageSize]) => {
+            // Paging, Sorting, Filter und Display Columns updaten
+            this.updateTableState(savedSettings, urlSettings, defaultPageSize);
+
+            // Row Detail Expander aktivieren
+            if (this.rowDetail && this.rowDetail.showToggleColumn) {
+              this.displayedColumns.splice(0, 0, 'rowDetailExpander');
+            }
+
+            // Selektierung der Rows aktivieren
+            if (this.listActions) {
+              this.displayedColumns.splice(0, 0, 'select');
+            }
+
+            // Selektierungs- und Row-Aktionen aktivieren
+            if (this.showListActions || this.rowActions) {
+              this.displayedColumns.push('options');
+            }
+
+            this.cd.markForCheck();
+            this.dataSource.updateData(false);
+          },
+          error: (err: Error | any) => {
+            this.dataSource.error = err;
+          },
+        })
+    );
+  }
+
+  private updateTableState(tableSettings: Partial<IPsTableSetting>, urlSettings: Partial<IPsTableUpdateDataInfo>, defaultPageSize: number) {
+    tableSettings = tableSettings || {};
+    urlSettings = urlSettings || {};
+
+    this.pageIndex = Math.max(urlSettings.currentPage, 0) || 0;

```

Comment:
Das || 0 hier brauchts nicht oder?

Reply:
War vermutlich wegen: Math.max(NaN, 0) -> NaN
Aber hab das jetzt in fromQueryParams gefixt, jetzt kann kein NaN mehr kommen.